### PR TITLE
fix: return non-zero exit code when matrix entries fail

### PR DESCRIPTION
**Pull request by eamonnmoloney**
## Summary

- `deploy-camunda matrix run` now exits with code 1 when any entry fails
- Previously, failed entries were logged and included in the summary but the process exited 0
- This caused CI steps to show as passed despite deployment timeouts/failures

## Root Cause

Both `runSequential()` and `runParallel()` only returned an error when `--stop-on-failure` was set. Otherwise, failed entries were collected in the results slice but `nil` was returned to the caller. Cobra's `RunE` propagated that `nil`, so `os.Exit(0)` was called.

## Fix

After `runSequential`/`runParallel` return, count failed results and synthesize a summary error:

```go
if retErr == nil {
    var failCount int
    for _, r := range results {
        if r.Error != nil {
            failCount++
        }
    }
    if failCount > 0 {
        retErr = fmt.Errorf("%d of %d matrix entries failed", failCount, len(results))
    }
}
```

## Impact

- CI steps using `deploy-camunda matrix run` will now correctly fail when deployments fail
- The summary is still printed before the error exit
- `--stop-on-failure` behavior is unchanged (still stops early on first failure)
- No change to dry-run or coverage modes

## Testing

- Unit test `TestRunReturnsErrorOnFailedEntries` added (3 sub-tests: all pass, one failure, all failures)
- Build and vet pass

### CODE REVIEW

File: scripts/deploy-camunda/matrix/runner.go [hunk 1]
```diff
@@ -258,6 +258,21 @@ func Run(ctx context.Context, entries []Entry, opts RunOptions) ([]RunResult, er
...
+		var failCount int
+		for _, r := range results {
+			if r.Error != nil {
+				failCount++
+			}
+		}
+		if failCount > 0 {
+			retErr = fmt.Errorf("%d of %d matrix entries failed", failCount, len(results))
+		}
+	}
```
[the remark this answers is not in the document]
Comment (eamonnmoloney) [reply scripts/deploy-camunda/matrix/runner.go:267]: Done in 13e5c0a. Extracted `synthesizeRunError(ctx, results, totalEntries)` as an unexported helper in `runner.go`. It now checks `ctx.Err()` when `len(results) < totalEntries` and returns a `run cancelled: N of M entries never started` error, covering the edge case where `runParallel` returns `(nil, nil)` because the parent context was already cancelled before any entry was dispatched.

File: scripts/deploy-camunda/matrix/matrix_test.go [hunk 2]
```diff
@@ -1311,6 +1312,74 @@ func sliceEqual(a, b []string) bool {
...
+			var failCount int
+			for _, r := range tt.results {
+				if r.Error != nil {
+					failCount++
+				}
+			}
+			if failCount > 0 {
+				retErr = fmt.Errorf("%d of %d matrix entries failed", failCount, len(tt.results))
+			}
+
```
[the remark this answers is not in the document]
Comment (eamonnmoloney) [reply scripts/deploy-camunda/matrix/matrix_test.go]: Done in 13e5c0a. The test is now `TestSynthesizeRunError` and calls the production `synthesizeRunError()` helper directly instead of re-implementing the logic. Also added two new test cases for context cancellation: zero results and partial results.

**Review by bkenez — APPROVED** [review]
